pkg/api/2019-10-27-preview/types.go

```diff
@@ -128,11 +128,19 @@ type RouterProfile struct {
 
 // MasterPoolProfile contains configuration for OpenShift master VMs.
 type MasterPoolProfile struct {
-	Count      *int64  `json:"count,omitempty"`
-	VMSize     *VMSize `json:"vmSize,omitempty"`
-	SubnetCIDR *string `json:"subnetCidr,omitempty"`
-	// PrivateAPIServer (temporary): Specifies if API server is public or private
+	Count         *int64         `json:"count,omitempty"`
+	VMSize        *VMSize        `json:"vmSize,omitempty"`
+	SubnetCIDR    *string        `json:"subnetCidr,omitempty"`
+	APIProperties *APIProperties `json:"apiProperties,omitempty"`
+}
+
+// APIProperties defines further properties on the API
+type APIProperties struct {
+	// PrivateAPIServer (in, optional): Specifies if API server is public or private
 	PrivateAPIServer *bool `json:"privateApiServer,omitempty"`
+
+	// VanityDomain (in, optional): define the custom root domain to use for the api server
+	VanityDomain *string `json:"vanityDomain,omitempty"`
```

Comment:
how does this field differ from properties.fqdn ?

Reply:
Not sure, :confused: I am inclined to remove it until we use it. (or i could just comment it out)

Reply:
This can be removed